### PR TITLE
Refactor deployed model labels

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
**What this PR does / why we need it**:

The labels implementation of Merlin resources (model services, batch jobs) is having two major limitations:
1. Hard coded to use "gojek.com/" as prefix so user can change it
2. Only propagate project's labels even though version labels are available via [new_model_version()](https://github.com/gojek/merlin/blob/main/python/sdk/merlin/fluent.py#L157) function.

These drawbacks limit the user to have more flexibility on using label for their tracking.

This PR make following changes:
1. Allow user to specify label prefix
2. Merge version labels with project labels. Version labels have more priority.
3. Introduce "component" labels to tag model-version, model-endpoint, batch-job, and image-builder resources.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/models/metadata.go [hunk 1]
```diff
@@ -30,7 +29,6 @@ const (
 	labelAppName          = "gojek.com/app"
```
Comment (pradithya) [inline api/models/metadata.go:29]: Why are these not configurable using the user prefix?
```
	labelAppName          = "gojek.com/app"
	labelOrchestratorName = "gojek.com/orchestrator"
	labelEnvironment      = "gojek.com/environment"
```

**Review by pradithya — CHANGES_REQUESTED** [review]
I think it'll be better to make the prefix consistent across all labels.

**Comment by pradithya** [comment]
Can you review https://github.com/gojek/merlin/pull/348 and rebase this PR against that. I think the e2e flakiness is due to kserve

File: api/models/metadata.go [hunk 2]
```diff
@@ -18,34 +18,47 @@ import (
...
+	LabelComponent        = "component"
+	LabelEnvironment      = "environment"
+	LabelOrchestratorName = "orchestrator"
+	LabelStreamName       = "stream"
+	LabelTeamName         = "team"
+
+	ComponentBatchJob      = "batch-job"
+	ComponentImageBuilder  = "image-builder"
+	ComponentModelEndpoint = "model-endpoint"
+	ComponentModelVersion  = "model-version"
```
Comment (ariefrahmansyah) [inline api/models/metadata.go:37]: I introduce a new label, called "component". Currently, only the image builder uses this component label, and this PR adds this component to other resources.

File: api/models/metadata.go (hunk 1 above)
Comment (ariefrahmansyah) [reply api/models/metadata.go]: Updated

File: api/models/metadata.go [hunk 3]
```diff
@@ -18,34 +18,47 @@ import (
...
+	LabelOrchestratorName: true,
+	LabelStreamName:       true,
+	LabelTeamName:         true,
+}
+
+var prefix string
+
+// InitKubernetesLabeller builds a new KubernetesLabeller Singleton
+func InitKubernetesLabeller(p string) {
+	prefix = p
```
Comment (tiopramayudi) [inline api/models/metadata.go:53]: can we do validation whether the prefix is valid or not? e.g "goto/gojek/" is not a valid prefix, if this value passed during init of merlin api, user won't be able to do anything

File: api/models/metadata.go (hunk 2 above)
Comment (pradithya) [reply api/models/metadata.go:37]: Noted!

File: api/models/metadata.go [hunk 4]
```diff
@@ -64,28 +77,33 @@ func (metadata *Metadata) Scan(value interface{}) error {
 
 func (metadata *Metadata) ToLabel() map[string]string {
 	labels := map[string]string{
-		labelOrchestratorName: "merlin",
-		labelAppName:          metadata.App,
-		labelEnvironment:      metadata.Environment,
-		labelStreamName:       metadata.Stream,
-		labelTeamName:         metadata.Team,
+		prefix + LabelAppName:          metadata.App,
```
Comment (tiopramayudi) [inline api/models/metadata.go:80]: does user need to explictily add `/` in the prefix?

**Review by pradithya — APPROVED** [review]
LGTM, thanks @ariefrahmansyah

File: api/models/metadata.go [hunk 5]
```diff
@@ -18,34 +18,47 @@ import (
...
 
 const (
-	labelTeamName         = "gojek.com/team"
-	labelStreamName       = "gojek.com/stream"
-	labelAppName          = "gojek.com/app"
-	labelOrchestratorName = "gojek.com/orchestrator"
-	labelEnvironment      = "gojek.com/environment"
-	labelUsersPrefix      = "gojek.com/%s"
+	LabelAppName          = "app"
+	LabelComponent        = "component"
```
Comment (tiopramayudi) [inline api/models/metadata.go:28]: do we need to expose this const?

File: api/models/metadata.go (hunk 4 above)
Comment (ariefrahmansyah) [reply api/models/metadata.go:80]: Yes, so it's up to user to set their own delimiter or not at all.

File: api/models/metadata.go (hunk 5 above)
Comment (ariefrahmansyah) [reply api/models/metadata.go:28]: Currently, only LabelOrchestratorName that used by other packages, so yeah, we will change other to local variables.

File: api/models/metadata.go (hunk 3 above)
Comment (ariefrahmansyah) [reply api/models/metadata.go:66]: Great suggestion. We added the validation for prefix.

File: api/models/metadata.go (hunk 4 above)
Comment (tiopramayudi) [reply api/models/metadata.go]: but the latest changes, we enforce user to add `/` on their prefix, since prefix "gojek.com" won't pass the validation

Comment (ariefrahmansyah) [reply api/models/metadata.go]: Correct. We've added the validation as you suggested when initializing the prefix to guard this enforcement.

**Review by tiopramayudi — APPROVED** [review]

**Comment by tiopramayudi** [comment]
lgtm

**Pull request merged**